raiden/transfer/channel.py

```diff
@@ -1837,6 +2087,20 @@ def apply_channel_newbalance(
         update_contract_balance(channel_state.partner_state, contract_balance)
 
 
+def handle_channel_withdraw(
+    channel_state: NettingChannelState, state_change: ContractReceiveChannelWithdraw
+) -> TransitionResult[NettingChannelState]:
+    if state_change.participant == channel_state.our_state.address:
+        end_state = channel_state.our_state
+    else:
+        end_state = channel_state.partner_state
+
+    end_state.total_withdraw = state_change.total_withdraw
```

Comment:
This is not correct. E.g.:

- A requests withdraw from B with `total_withdraw=10`
- B confirms withdraw for A with `total_withdraw=10`
- A sends withdraw to the chain
- A requests withdraw from B with `total_withdraw=20`
* The transaction for the first withdraw is confirmed, this will overwrite the `total_withdraw`
- B confirms the `total_withdraw=20`, A will wrongly reject

Reply:
The value confirmed on-chain and the value request off-chain must be separated.

Reply:
Also, the value has to increase, if the value is lower it should be ignored.